lib/chef/provider/package/windows.rb

```diff
@@ -129,6 +129,7 @@ def action_install
         # Chef::Provider::Package action_install + action_remove call install_package + remove_package
         # Pass those calls to the correct sub-provider
         def install_package(name, version)
+          fail Chef::Exceptions::NoWindowsPackageSource, "installer_type '#{installer_type}' requires a source" if new_resource.source.nil?
```

Comment:
> Not sure I get your comment about the fix being in the cookbook. The goal here is to fix core assuming the cookbook is not used.

The fix in behavior, e.g., making it idempotent is here. but if the cookbook didn't require source (apparently it does), and core did, the most pain would be in the transition for existing users. So that's the biggest band-aid/fix/hug. It's kind of moot of the behavior is fairly similar. We probably should have made a behavior matrix somewhere on paper earlier in the week.